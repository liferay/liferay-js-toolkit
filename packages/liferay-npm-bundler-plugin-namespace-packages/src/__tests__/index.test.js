import PkgDesc from 'liferay-npm-build-tools-common/lib/pkg-desc';
import PluginLogger from 'liferay-npm-build-tools-common/lib/plugin-logger';
import readJsonSync from 'read-json-sync';
import plugin from '../index';

const fixturesDir =
	`${process.cwd()}/packages/` +
	`liferay-npm-bundler-plugin-namespace-packages/` +
	`src/__tests__/project`;

it('namespaces packages correctly for the root package', () => {
	const rootPkgJson = readJsonSync(`${fixturesDir}/package.json`);
	const dir = fixturesDir;
	const pkgJson = readJsonSync(`${dir}/package.json`);
	const pkg = new PkgDesc(pkgJson.name, pkgJson.version, fixturesDir, true);
	const log = new PluginLogger();

	plugin({pkg, log, rootPkgJson}, {pkgJson});

	expect(pkgJson).toMatchSnapshot();
});

it('namespaces packages correctly for non-root package', () => {
	const rootPkgJson = readJsonSync(`${fixturesDir}/package.json`);
	const dir = `${fixturesDir}/node_modules/is-finite`;
	const pkgJson = readJsonSync(`${dir}/package.json`);
	const pkg = new PkgDesc(pkgJson.name, pkgJson.version, dir);
	const log = new PluginLogger();

	plugin({pkg, log, rootPkgJson}, {pkgJson});

	expect(pkgJson).toMatchSnapshot();
});

it('logs results correctly', () => {
	const rootPkgJson = readJsonSync(`${fixturesDir}/package.json`);
	const dir = `${fixturesDir}/node_modules/is-finite`;
	const pkgJson = readJsonSync(`${dir}/package.json`);
	const pkg = new PkgDesc(pkgJson.name, pkgJson.version, dir);
	const log = new PluginLogger();

	plugin({pkg, log, rootPkgJson}, {pkgJson});

	expect(log.messages).toMatchSnapshot();
});
