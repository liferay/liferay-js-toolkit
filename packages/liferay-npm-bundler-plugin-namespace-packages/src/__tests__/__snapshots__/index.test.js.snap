// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs results correctly 1`] = `
Array [
  Object {
    "level": "info",
    "source": "namespace-packages",
    "things": Array [
      "Namespaced package",
    ],
  },
  Object {
    "level": "info",
    "source": "namespace-packages",
    "things": Array [
      "Namespaced",
      1,
      "dependencies",
    ],
  },
]
`;

exports[`namespaces packages correctly for non-root package 1`] = `
Object {
  "dependencies": Object {
    "project2$number-is-nan": "1.0.0",
  },
  "name": "project2$is-finite",
  "version": "1.0.0",
}
`;

exports[`namespaces packages correctly for the root package 1`] = `
Object {
  "dependencies": Object {
    "project2$is-finite": "1.0.0",
    "project2$isobject": "1.0.0",
  },
  "name": "project2",
  "version": "1.0.0",
}
`;
